### PR TITLE
FIX: Build docs with branches of forks too

### PR DESCRIPTION
**Pull request by MridulS**
Let's see if this fixes https://github.com/scipp/copier_template/issues/151

### CODE REVIEW

**Comment by MridulS** [comment]
Yeah this fixes building from forks

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -50,6 +50,7 @@ jobs:
       - uses: actions/checkout@v3
         with:
           ref: ${{ inputs.branch == '' && github.ref_name || inputs.branch }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```
Comment (nvaytet) [inline .github/workflows/docs.yml:53]:
```suggestion
          # see https://github.com/actions/checkout/issues/551#issuecomment-1113089390
          repository: ${{ github.event.pull_request.head.repo.full_name }}
```

Comment (MridulS) [reply .github/workflows/docs.yml:53]: Shouldn't this be synced up with copier template too then?

**Review by nvaytet — APPROVED** [review]

File: .github/workflows/docs.yml (hunk 1 above)
Comment (nvaytet) [reply .github/workflows/docs.yml:53]: This was a suggestion, you can of course take it or leave it.

**Pull request merged**